gh-1294 added discovery_set to vshard (#1376)
tarantool/doc: doc/reference/reference_rock/vshard/vshard_ref.rst

    Force wakeup of the bucket discovery fiber.

.. _router_api-discovery_set:

.. function:: vshard.router.discovery_set(mode)

    Turn on/off the background discovery fiber used by the router to
    find buckets.

    :param mode: working mode of a discovery fiber. There are three modes: ``on``,
                 ``off`` and ``once``

>     When the mode is ``on`` (default), the discovery fiber works all the lifetime

Reviewer: ```suggestion
    When the mode is ``on`` (default), the discovery fiber works during all the lifetime
```

    of the router. Even after all buckets are discovered, it will

>     still come to storages and download their buckets with some big

Reviewer: Would be nice to clarify the rules for setting the period. There'll be questions.

Author: fixed in c826865

>     period. This is useful if a bucket topology changes often and the number of

Reviewer: ```suggestion
    period. This is useful if the bucket topology changes often and the number of
```

    buckets is not big. The router will keep its route table up to
    date even when no requests are processed.

>     When the mode is ``off``, a discovery is disabled completely.

Reviewer: ```suggestion
    When the mode is ``off``, discovery is disabled completely.
```

>     When the mode is ``once``, a discovery will start, find locations of

Reviewer: ```suggestion
    When the mode is ``once``, discovery will start, find the locations of
```

    all buckets, and then the discovery fiber is terminated. This

>     is good for large bucket count and for clusters, where rebalancing is rare.

Reviewer: ```suggestion
    is good for a large bucket count and for clusters, where rebalancing is rare.
```

    The method is good to enable/disable discovery after the router is
    already started, but discovery is enabled by default. You may want

>     to never enable it even for a short time - then specify

Reviewer: ```suggestion
    to never enable it even for a short time -- then specify the
```

>     discovery_mode option in the :ref:`configuration <cfg_basic-discovery_set>`.

Reviewer: ```suggestion
    ``discovery_mode`` option in the :ref:`configuration <cfg_basic-discovery_set>`.
```

    It takes the same values as :samp:`vshard.router.discovery_set({mode})`.

    You may decide to turn off discovery or make it ``once`` if you have
    many routers, or tons of buckets (hundreds of thousands and more),
    and you see that the discovery process consumes notable CPU % on
    routers and storages. In that case it may be wise to turn off the
    discovery when there is no rebalancing in the cluster. And turn it

>     on for new routers, as well as for all routers when a rebalancing is

Reviewer: ```suggestion
    on for new routers, as well as for all routers when rebalancing is
```

    Force wakeup of the bucket discovery fiber.

.. _router_api-discovery_set:

.. function:: vshard.router.discovery_set(mode)

    Turn on/off the background discovery fiber used by the router to
    find buckets.

    :param mode: working mode of a discovery fiber. There are three modes: ``on``,
                 ``off`` and ``once``

    When the mode is ``on`` (default), the discovery fiber works during all the lifetime
    of the router. Even after all buckets are discovered, it will
    still come to storages and download their buckets with some big
    period (`DISCOVERY_IDLE_INTERVAL <https://github.com/tarantool/vshard/blob/master/vshard/consts.lua>`_).
    This is useful if the bucket topology changes often and the number of
    buckets is not big. The router will keep its route table up to
    date even when no requests are processed.

    When the mode is ``off``, discovery is disabled completely.

>     When the mode is ``once``, discovery will start, find the locations of

Reviewer: ```suggestion
    When the mode is ``once``, discovery starts and finds the locations of
```

    all buckets, and then the discovery fiber is terminated. This
    is good for a large bucket count and for clusters, where rebalancing is rare.

    The method is good to enable/disable discovery after the router is
    already started, but discovery is enabled by default. You may want
    to never enable it even for a short time -- then specify the

>     ``discovery_mode`` option in the :ref:`configuration <cfg_basic-discovery_set>`.

Reviewer: let's add `discovery-mode` to the list of options here, like we do for `discovery_set`. otherwise the reference feels "patchy".

Author: my mistake. There is the only cfg option -- discovery_mode. Fixed in 3d2f7ac
